### ISSUE TITLE
Add warning about multi-level break gotcha under sh/bash

### ISSUE THREAD

**Issue by drvink** [koalaman/shellcheck#1983]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
wacky()
{
    printf 'a\nb\nc\n' | while read -r blah; do
        echo "$blah" >/dev/null
        n=1
        while [ $((n)) -le 3 ]; do
            echo "$n"
            printf "one\ntwo\nthree\n" | while read -r resp; do
                if [ "$resp" = two ]; then
                    break 2
                fi
            done
            n=$((n + 1))
        done
    done
}

wacky
```

prints:

```
1
2
3
1
2
3
1
2
3
```

#### Here's what shellcheck currently says:
(it says nothing)

#### Here's what I wanted or expected to see:
Shellcheck should have a warning about this, because the behavior is rather surprising/unexpected. According to [the text of IEEE 1003.1-2007](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#break), it would seem that the `break 2` above should exit the outermost loop, but in at least two popular Bourne-ish shells (I tried dash and bash), it does not. In zsh, it does!
